variables.tf

```diff
@@ -210,6 +211,13 @@ variable "vmware_vmm_domains" {
     ])
     error_message = "`resolution_immediacy`: Allowed values are `immediate`, `lazy` or `pre-provision`."
   }
+
+  validation {
+    condition = alltrue([
+      for dom in var.vmware_vmm_domains : dom.custom_epg_name == null || try(can(regex("^[a-zA-Z0-9_.-]{0,64}$", dom.custom_epg_name)))
```

Comment:
The fallback value for try should be false: `try(can(regex("^[a-zA-Z0-9_.-]{0,64}$", dom.custom_epg_name)), false)`

Reply:
Did some additional testing.
I think that we only need try when we validate with contains(), because contains returns error when the variable under test is null:
```
> contains(["equals"], null)

> ╷
│ Error: Invalid function argument
│
│   on <console-input> line 1:
│   (source code not available)
│
│ Invalid value for "value" parameter: argument must not be null.
╵
```

regex() returns false when the variable under test is null:
```
> can(regex("regex123", null))
false
```
So I think this should be enough:
```
dom.custom_epg_name == null || can(regex("^[a-zA-Z0-9_.-]{0,64}$", dom.custom_epg_name))
```
What do you think?

Reply:
I've checked the documentation once again:
_can() evaluates the given expression and returns a boolean value indicating whether the expression produced a result without any errors._
```
> regex("regex123", null)

> ╷
│ Error: Invalid function argument
│
│   on <console-input> line 1:
│   (source code not available)
│
│ Invalid value for "string" parameter: argument must not be null.
╵

>

> can(regex("regex123", null))
false
```

So I think we do not need try() at all when we already use can():
```
dom.custom_epg_name == null || can(regex("^[a-zA-Z0-9_.-]{0,64}$", dom.custom_epg_name))
```
What do you think?